r/R/dataset-format.R

```diff
@@ -101,8 +101,10 @@ IpcFileFormat <- R6Class("IpcFileFormat", inherit = FileFormat)
 #' @rdname FileFormat
 #' @export
 CsvFileFormat <- R6Class("CsvFileFormat", inherit = FileFormat)
-CsvFileFormat$create <- function(..., opts = csv_file_format_parse_options(...)) {
-  dataset___CsvFileFormat__Make(opts)
+CsvFileFormat$create <- function(...,
+                                 opts = csv_file_format_parse_options(...),
+                                 compression = CompressionType$UNCOMPRESSED) {
+  dataset___CsvFileFormat__Make(opts, compression)
```

Comment:
We have a helper used in a few places that makes this a little more human-friendly:

```suggestion
                                 compression = "uncompressed") {
  dataset___CsvFileFormat__Make(opts, compression_from_name(compression))
```

Should also add a note in the FileFormat$create docs above (around L48) that compression is an option now.